argyrogounari/assignment2/BinaryTree.java

```diff
@@ -0,0 +1,101 @@
+
+/**
+* Class representing an arbitrary Binary Tree.
+*
+* @author  Argyro Gounari
+* @version 1.1
+* @since   29-05-2018 
+*/
+public class BinaryTree<T> {
+
+	private Node<T> root;
+	
+	/**
+	* Creates a binary tree. 
+	*
+	* @param key
+	* @param value
+	*/
+	public void addNode(int key, T value) {
+		
+		Node<T> newNode = new Node<T>(key, value);
+		
+		if (root == null) {
+
+			root = newNode;
+
+		} else {
+			
+			Node<T> focusNode = root;
+			Node<T> parent;
+
+			while (true) {
+				
+				parent = focusNode;
+
+				if (parent.getLeft() == null) {
+					
+					parent.setLeft(newNode); 
+					return; 
+					
+				} else if(parent.getRight() == null){
+					
+					parent.setRight(newNode);
+					return; 
+
+				}
+				
+				focusNode = parent.getRight(); //stores only on the right
+				
+			}
+		}
+
+	}
+
+	/**
+	* Q1 - Print Ancestors
+	* Prints all the ancestors of the key 
+	* in the given binary tree.
+	*
+	* @param node
+	* @param key
+	*/
+	public boolean printAncestors(Node<T> node, int key) {
```

Comment:
It can be useful to have a helper method that printAncestors() uses which returns a list of ancestor keys. Then this helper can be used in tests where you can assert that the list outputed matches what you expect.